internal/cloud/backend_test.go

```diff
@@ -273,340 +273,43 @@ func TestCloud_PrepareConfigWithEnvVars(t *testing.T) {
-				"workspaces": cty.NullVal(cty.Object(map[string]cty.Type{
-					"name":    cty.String,
-					"tags":    cty.Set(cty.String),
-					"project": cty.String,
-				})),
-			}),
-			vars: map[string]string{
-				"TF_WORKSPACE": "i-dont-exist-in-org",
-			},
-			expectedErr: `Invalid workspace selection: OpenTofu failed to find workspace "i-dont-exist-in-org" in organization hashicorp`,
-		},
-		"workspaces and env var specified": {
-			config: cty.ObjectVal(map[string]cty.Value{
-				"hostname":     cty.NullVal(cty.String),
-				"token":        cty.NullVal(cty.String),
-				"organization": cty.StringVal("mordor"),
-				"workspaces": cty.ObjectVal(map[string]cty.Value{
-					"name":    cty.StringVal("mt-doom"),
-					"tags":    cty.NullVal(cty.Set(cty.String)),
-					"project": cty.NullVal(cty.String),
-				}),
-			}),
-			vars: map[string]string{
-				"TF_WORKSPACE": "shire",
-			},
-			expectedWorkspaceName: "mt-doom",
-		},
-		"env var workspace does not have specified tag": {
-			setup: func(b *Cloud) {
-				b.client.Organizations.Create(context.Background(), tfe.OrganizationCreateOptions{
-					Name: tfe.String("mordor"),
-				})
-
-				b.client.Workspaces.Create(context.Background(), "mordor", tfe.WorkspaceCreateOptions{
-					Name: tfe.String("shire"),
-				})
-			},
-			config: cty.ObjectVal(map[string]cty.Value{
-				"hostname":     cty.NullVal(cty.String),
-				"token":        cty.NullVal(cty.String),
-				"organization": cty.StringVal("mordor"),
-				"workspaces": cty.ObjectVal(map[string]cty.Value{
-					"name": cty.NullVal(cty.String),
-					"tags": cty.SetVal([]cty.Value{
-						cty.StringVal("cloud"),
-					}),
-					"project": cty.NullVal(cty.String),
-				}),
-			}),
-			vars: map[string]string{
-				"TF_WORKSPACE": "shire",
-			},
-			expectedErr: "OpenTofu failed to find workspace \"shire\" with the tags specified in your configuration:\n[cloud]",
-		},
-		"env var workspace has specified tag": {
-			setup: func(b *Cloud) {
-				b.client.Organizations.Create(context.Background(), tfe.OrganizationCreateOptions{
-					Name: tfe.String("mordor"),
-				})
-
-				b.client.Workspaces.Create(context.Background(), "mordor", tfe.WorkspaceCreateOptions{
-					Name: tfe.String("shire"),
-					Tags: []*tfe.Tag{
-						{
-							Name: "hobbity",
-						},
-					},
-				})
-			},
-			config: cty.ObjectVal(map[string]cty.Value{
-				"hostname":     cty.NullVal(cty.String),
-				"token":        cty.NullVal(cty.String),
-				"organization": cty.StringVal("mordor"),
-				"workspaces": cty.ObjectVal(map[string]cty.Value{
-					"name": cty.NullVal(cty.String),
-					"tags": cty.SetVal([]cty.Value{
-						cty.StringVal("hobbity"),
-					}),
-					"project": cty.NullVal(cty.String),
-				}),
-			}),
-			vars: map[string]string{
-				"TF_WORKSPACE": "shire",
-			},
-			expectedWorkspaceName: "", // No error is raised, but workspace is not set
-		},
-		"project specified": {
+		"with workspace env var overwrite tags config value": {
```

Comment:
Maybe I missed something here, but shouldn't this test cause an error? If workspace overwrite is not in the tags, then there should be an error

There's no `expectedErr` in this test case

Reply:
Well spotted! Thank you Arel! I fixed the test definition. Unfortunately, I feel like most of the tests (no relevant to this issue) with inherited validation will require thorough cross-check as a followup.